grails-app/services/org/pih/warehouse/product/ProductService.groovy

```diff
@@ -1248,40 +1249,53 @@ class ProductService {
                 productInstance.save(flush: true, failOnError: true)
                 if(productJson.containsKey("attributes")){
                     JSONArray attributesJsonArray = productJson.getJSONArray("attributes")
+                    log.info "Attributes json:${attributesJsonArray?.toString()}"
                     for(int i=0;i<attributesJsonArray.length();i++){
                         JSONObject attributeJson = attributesJsonArray.getJSONObject(i)
+                        log.info "Attribute Object json:${attributeJson?.toString()}"
                         Attribute attribute = Attribute.findByCode(attributeJson.getString("code"))
```

Comment:
Let's assign the code to a variable and use that throughout rather than getting the string out of the json over and over.
```
String attributeCode = attributeJson.getString("code")
Attribute attribute = Attribute.findByCode(attributeCode)
```